lib/add_to_wallet.dart

```diff
@@ -1,10 +1,39 @@
 import 'dart:async';
-import 'dart:io';
 
 import 'package:flutter/services.dart';
 
 export 'widgets/add_to_wallet_button.dart';
 
+
 class AddToWallet {
   static const MethodChannel _channel = const MethodChannel('add_to_wallet');
+
+  static final AddToWallet _instance = AddToWallet._internal();
+
+  /// Associate each Widget to its click event handlers
+  static final Map<String, FutureOr<dynamic> Function(MethodCall)> _handlers = Map();
+
+  factory AddToWallet() {
+    return _instance;
+  }
+
+  AddToWallet._internal() {
+    initMethodCallHandler();
+  }
+
+  void initMethodCallHandler() => _channel.setMethodCallHandler(_handleCalls);
```

Comment:
This should be private 
```suggestion
  void _initMethodCallHandler() => _channel.setMethodCallHandler(_handleCalls);
```